app/models/spree/payment_decorator.rb

```diff
@@ -1,9 +1,13 @@
 Spree::Payment.class_eval do
-  before_update :send_gift_card, :if => "state_changed? && state_was != 'completed' && state == 'completed'"
+  after_update :send_gift_card, if: :gift_card_valid_state_transition?
```

Comment:
@sawangupta92 - We can also do this by state_machine after_transition, what say?